### PR TITLE
Learning goals

### PR DESCRIPTION
**Pull request by ViktoriyaHaiduk**
---
name: solution review 
about: A template PR for code review with a checklist
---

### Files

- [ ] The file name describes the function's behavior
- [ ] There is a module docstring in the function file
- [ ] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [ ] There is a module docstring in the tests file

### Unit Tests

- [ ] The test class has a helpful name in PascalCase
- [ ] The test class has a docstring
- Every unit test has
  - [ ] A helpful name
  - [ ] A clear docstring
  - [ ] Only one assertion
  - [ ] There is no logic in the unit test
- [ ] All tests pass
- [ ] There are tests for defensive assertions
- [ ] There are tests for boundary cases

### Function Docstring

- [ ] The function's behavior is described
- The function's arguments are described:
  - [ ] Type
  - [ ] Purpose
  - [ ] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [ ] Type
  - [ ] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [ ] Each assumption about an argument is checked with an assertion
  - [ ] Each assertion checks for _only one_ assumption about the argument
- [ ] Include 3 or more (passing!) doctests

### The Function

- [ ] The function's name describes it's behavior
- [ ] The function's name matches the file name
- [ ] The function has correct type annotations

### Do's

- [ ] Variable names help to understand the strategy
- [ ] Any comments are clear and describe the strategy
- [ ] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [ ] The function's strategy _is not_ described in the documentation
- [ ] Comments explain the _strategy_, **not** the _implementation_
- [ ] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [ ] The code passes the formatting checks
- [ ] The code passes all Ruff linting checks
- [ ] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [ ] Variables are named with snake_case
- [ ] Variable names are clear and helpful
- [ ] The code follows the strategy as simply as possible
- [ ] The implementation is as simple as possible given the strategy
- [ ] The code includes defensive assertions
- [ ] Defensive assertions include as little logic as possible

### CODE REVIEW

File: collaboration/learning_goals.md [hunk 1]
```diff
@@ -1,5 +1,39 @@
...
+- Communicate constructively and provide actionable feedback to team members. 🗣️  
+- Promote a culture of collaboration and shared learning to boost overall team performance. 🌱  
+
+### Conduct Peer Code Reviews 👀  
+
+- Review team members’ code solutions for quality, functionality, and adherence to standards. ✔️  
+- Execute tests and debugging processes to identify issues and discuss improvements effectively. 🐞  
+
+## Individual Goals 🎯
+
```
Comment (linahKhayri) [inline collaboration/learning_goals.md:25]: You did a great job adding these points. **One suggestion** I have is to include a point about adopting TDD when writing code as this could help ensure more accurate tests and reduce the chances of bugs.

**Review by linahKhayri — COMMENTED** [review]
Good job guys! the document is well structured and neat, easy to read and follow and you provide solid points.
I just have this small suggestion I added in the individual goals section :)
_Looking forward to see more of your amazing work in Future_

File: collaboration/learning_goals.md (hunk 1 above)
Comment (ViktoriyaHaiduk) [reply collaboration/learning_goals.md:30]: Thank you @linahKhayri. Added.

**Review by linahKhayri — DISMISSED** [review]
Well done! You did an amazing job

File: collaboration/learning_goals.md [hunk 2]
```diff
@@ -1,5 +1,49 @@
...
+- Develop the ability to provide and apply constructive feedback in code reviews. 💬  
+- Enhance skills in writing clean, tested, and well-documented code. ✍️  
 
-## Individual
+## Collective Goals 🌟
+
+### Master Collaborative Git Workflows 🚀  
+
+- Push markdown files and solutions to repositories following best practices. 📂  
+- Manage pull requests, resolve conflicts, and adhere to team agreements🔄
```
Comment (Khusro-S) [inline collaboration/learning_goals.md:12]: I think it might be helpful to elaborate bit more on these lines, Im suggesting these down below:

```suggestion
### Push Markdown Files and Solutions📂:
- Learn how to organize and structure Markdown files effectively for documentation📝.
- Practice pushing your work to the repository following best practices, such as clear commit messages and proper folder structures✅.

### Manage Pull Requests🔀:
- Create pull requests to propose changes to the main repository📤.
- Review and respond to feedback on your pull requests💬.

### Resolve Merge Conflicts⚔️:
- Understand why merge conflicts occur and how to resolve them using Git tools🤔.
- Learn techniques to minimize the likelihood of conflicts, such as pulling the latest changes before committing🔄.

### Follow Team Agreements🤝:
- Adhere to guidelines, conventions and norms/rules set by the team, naming branches for example📋.
- Collaborate effectively while respecting team decisions and workflows🌟.
```

**Review by Khusro-S — CHANGES_REQUESTED** [review]
Great work overall! I really appreciate the effort you've put into writing these goals. I just have a couple of suggestions that might add a bit more clarity and make the goals even more beginner-friendly

**Review by Khusro-S — APPROVED** [review]
That's great. Everything seems perfect! Great job everyone 🎉.

**Review by linahKhayri — APPROVED** [review]
Good job again guys!

**Comment by ViktoriyaHaiduk** [comment]
Merging with main branch
